### PR TITLE
Fabric: Update for 1.17

### PR DESCRIPTION
**Pull request by MeeniMc**
This is simply an update of Loom/fabric/gradle for 1.17.  This also changes version and icons to make modmenu happy.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,4 @@
 Removes need to have an arrow in your inventory to use the Infinity I enchant on your bow.
 
 **Thanks to Myrathi for the original InfiniBows!**
+**Thanks to MeeniMc for the 1.17 port**
```
Comment (Parker8283) [inline README.md]: As a general rule of thumb, I think I would leave stuff like this to the repo maintainer to add if they so desire. Some might view this the wrong way. Also, people can see the commit logs and see that you are essentially thanking yourself ;)

File: build.gradle [hunk 2]
```diff
@@ -14,20 +14,9 @@ minecraft {}
...
-
-processResources {
-    inputs.property "version", project.version
-    from(sourceSets.main.resources.srcDirs) {
-        include "fabric.mod.json"
-        expand "version": project.version
-    }
-    from(sourceSets.main.resources.srcDirs) {
-        exclude "fabric.mod.json"
-    }
```
Comment (Parker8283) [inline build.gradle:30]: Any particular reason why this was moved down?

File: build.gradle [hunk 3]
```diff
@@ -37,18 +26,25 @@ tasks.withType(JavaCompile) {
...
+}
 jar {
     manifest {
         attributes([
             "Specification-Title": "bowinfinityfix",
             "Specification-Version": "1",
             "Specification-Vendor": "Parker Young",
             "Implementation-Title": project.name,
-            "Implementation-Version": project.version,
+            "Implementation-Version": project.mod_version,
```
Comment (Parker8283) [inline build.gradle]: Why the move away from `project.version`?

File: gradle.properties [hunk 4]
```diff
@@ -2,14 +2,15 @@
 org.gradle.jvmargs=-Xmx1G
 
 # Mod Properties
-mod_version=rv9
+mod_version=1.10+1.17-fabric
```
Comment (Parker8283) [inline gradle.properties]: I would like to stick with my versioning scheme. I understand it's not standard, I know semantic versioning, I use it with basically all of my other projects. I'm just stubborn :P

This project is kind of old so I didn't follow it back then when I wrote it. And it would slightly irk me to see the version scheme change in the list of versions on CF. I know, not great reasons :) But I would prefer it that way.

Also, I don't like having the MC version in the mod version, just feel it's too busy. That's why I had the MC version in the `archiveAppendix` in the `build.gradle` (also because I couldn't figure out a good way to use variables to sub in the MC version, I hardcoded it)

File: src/main/resources/fabric.mod.json [hunk 5]
```diff
@@ -4,17 +4,18 @@
   "version": "${version}",
   "name": "Bow Infinity Fix",
   "description": "Removes need to have an arrow in your inventory to use the Infinity I enchant on your bow.",
-  "authors": ["Parker8283"],
+  "authors": ["Parker8283","MeeniMc"],
```
Comment (Parker8283) [inline src/main/resources/fabric.mod.json]: See `README` comment

File: src/main/resources/fabric.mod.json [hunk 6]
```diff
@@ -4,17 +4,18 @@
...
     "issues": "https://github.com/Parker8283/BowInfinityFix/issues"
   },
   "license": "LGPLv3",
+  "icon": "assets/bif/bif-logo-light.png",
   "environment": "*",
   "mixins": ["bif.mixins.json"],
   "depends": {
-    "fabricloader": ">=0.4.0",
-    "minecraft": "1.16.x"
+    "fabricloader": ">=0.11.6",
```
Comment (Parker8283) [inline src/main/resources/fabric.mod.json]: From what I remember (and correct me if I'm wrong), the Fabric Loader is independent of the MC version. So realistically, someone could use newer loader versions on older MC versions, and vice versa. Now, whether or not that is advisable is another question :P

So, my question, is there anything from newer loader versions that I utilize that would require me to _need_ to uptick this?

File: src/main/resources/fabric.mod.json [hunk 7]
```diff
@@ -4,17 +4,18 @@
...
   "description": "Removes need to have an arrow in your inventory to use the Infinity I enchant on your bow.",
-  "authors": ["Parker8283"],
+  "authors": ["Parker8283","MeeniMc"],
   "contact": {
     "homepage": "https://www.curseforge.com/minecraft/mc-mods/bow-infinity-fix",
     "sources": "https://github.com/Parker8283/BowInfinityFix",
     "issues": "https://github.com/Parker8283/BowInfinityFix/issues"
   },
   "license": "LGPLv3",
+  "icon": "assets/bif/bif-logo-light.png",
```
Comment (Parker8283) [inline src/main/resources/fabric.mod.json]: The logos are quality programmer art :)

Real question, do I really _need_ logos? I've never used modmenu so I'm not sure what it looks like without it. Does the menu look ugly or something without a logo? I can't imagine I'm the only mod that is super small/insignificant, and if they essentially ~~require~~ "strongly recommend" logos, I personally think that is an oversight on their part. I honestly don't feel like this mod is worthy of a logo, it's too small, haha. Though I do appreciate the effort :)

File: build.gradle [hunk 8]
```diff
@@ -37,18 +26,25 @@ tasks.withType(JavaCompile) {
...
             "Implementation-Vendor": "Parker8283",
             "Implementation-Timestamp": new Date().format("yyyy-MM-dd'T'HH:mm:ssZ")
         ])
     }
     from "LICENSE"
-    archiveAppendix.set "1.16.x-fabric"
+    archiveAppendix.set "fabric"
```
Comment (Parker8283) [inline build.gradle]: See above comment on versioning

**Review by Parker8283 — CHANGES_REQUESTED** [review]
Hey, thanks for the PR! I normally wait to do this in one sitting once both Forge and Fabric have their versions released, because yada yada no time yada yada adult now :) So you jumped my gun! But that's cool, glad to see people actually care about this mod, haha.

Some comments on your PR. My questions are almost all actual genuine questions. I basically don't use Fabric at all (actually, I rarely play MC anymore), so I'm probably ignorant of its ins-and-outs. I only started providing a Fabric version because it was simple enough and figured, if people wanted to use Fabric, why not?

File: README.md (hunk 1 above)
Comment (MeeniMc) [reply README.md]: In my professional setting, the person that makes the change also update the copyrights/authors etc, so that it doesn't get lost/falls on somebody else's plate to update it. Our lawyers care quite a bit that these are maintained accurate so better doing it now than forget later. 

Now this is not the same here, we have no lawyers and everybody is pro-bono, I have no problem removing all this if that rubs you the wrong way :)

File: src/main/resources/fabric.mod.json (hunk 7 above)
Comment (MeeniMc) [reply src/main/resources/fabric.mod.json]: Mods with no logo all have the 'default' logo: a question mark on a gray background. The mod certainly works without a logo, I just though why not and used some mad powerpoint skills.

File: src/main/resources/fabric.mod.json (hunk 6 above)
Comment (MeeniMc) [reply src/main/resources/fabric.mod.json]: There is a minimum version of fabric loader that supports 1.17. I know for a fact that 0.11.3 works, and maybe some prior 0.11.x also work. 0.4.0 absolutely doesn't. The  change in Gradle is important because it is needed to have `gradle runClient` working; this one is not strictly needed since the predicate will necessarily be satisfied for any 1.17 deployment.

File: gradle.properties (hunk 4 above)
Comment (MeeniMc) [reply gradle.properties]: Starting the version number with 'rv' causes a warning to be produced in the game log (from the fabric loader I think). If there were no runtime warnings I would not have changed it (since this mod has no API to speak of). 

```
[main/WARN]: Mod `bowinfinityfix` (rv9) does not respect SemVer - comparison support is limited.
```

Up to you to decide if silencing that warning is enough to change your mind

File: README.md (hunk 1 above)
Comment (Parker8283) [reply README.md]: Huh, really? I guess we hang with different crowds :)

I guess it just seems kinda self-serving to me. I'm sure you didn't mean it that way. I get the professional setting thing, at my workplace there's about 30 copyright notices on each file, haha. But for MC mods, I think the rules are probably a little more lax. I guess my experience comes from someone who updated a FOSS project I used for something, and plastered their name absolutely everywhere, whereas the real authors just had like two sentences in the README. _That_ rubbed me the wrong way.

I guess for me, it's easier to enforce totally objective rules (like "the maintainer will be the one to update credits, etc") than work with more subjective ones (like "oh its okay for you to do it, but not this person because of my own arbitrary ideas on what constitutes creditable work"). Might make some more people upset, but they can't argue with it because there's little to no subjectivity involved.

So, TLDR, I would appreciate the removal of this line. Not because I'm trying to discredit your work. Because I'm weird :)

File: gradle.properties (hunk 4 above)
Comment (Parker8283) [reply gradle.properties]: Yeah, I've seen that line. One line in a log file that most users will not read isn't enough to sway me :)

From what I can tell, this deals with like update checking or ensuring client<->server are network compatible, or something like that? This mod almost always only gets one version per major MC version, so that doesn't matter to me.

File: src/main/resources/fabric.mod.json (hunk 7 above)
Comment (Parker8283) [reply src/main/resources/fabric.mod.json]: In my personal opinion, that quirk of the modmenu should be addressed. There should be a way to not render anything if a mod doesn't want a logo, [kinda like what Forge does by default](https://user-images.githubusercontent.com/5672271/123340867-915f4680-d512-11eb-837c-fc8a6e09b67b.png). But I guess that's not something I can fix :)

In that case, I can stick with these. But inspired by the fact that both you and I seem to be masters of programmer art, I just commissioned my sibling to make a proper logo for the mod. So apologies if I wipe out your logos later. Nothing personal :), but my CF page has been lacking in the logo department as well (I just literally ripped a game asset because CF doesn't let you post a mod without a logo...)

File: src/main/resources/fabric.mod.json (hunk 6 above)
Comment (Parker8283) [reply src/main/resources/fabric.mod.json]: Huh, the more you know. Alrighty, then this is fine. I just try not to uptick minimum required versions unless absolutely necessary. Mostly from my experience in the wider software industry, and people's affinity to stick with old versions of software :P

File: src/main/resources/fabric.mod.json (hunk 7 above)
Comment (MeeniMc) [reply src/main/resources/fabric.mod.json]: I have removed the logo as well from this pr, if you do finally want to use it, the material remains available in a separate branch that I have not PR'd here 
https://github.com/MeeniMc/BowInfinityFix/tree/pr/1.17

Feel free to pick it if you want.

File: build.gradle (hunk 8 above)
Comment (MeeniMc) [reply build.gradle]: This appendix get added only to the -dev.jar, the normal jar is simply named modname-version.jar (so without 1.17.x-fabric).

Comment (Parker8283) [reply build.gradle]: Oh you know what, you're right! Haha, I totally forgot about that. I spent a while trying to get it to work, but just gave up and manually renamed the built jar before uploading it to CF. Good times...alright then this is fine.

**Comment by Parker8283** [comment]
Coolio, I'll give this a test sometime this weekend, assuming I get the time. Since this is just a buildscript uptick, I imagine it's fine.

**Review by Parker8283 — APPROVED** [review]

**Comment by Parker8283** [comment]
Manually pulled this because I did other stuff after pulling yours locally for testing. Wish GH had a way to denote that without closing the normal way, but oh well. Thanks again!
